samples/client/petstore/C/model/api_response.c

```diff
@@ -19,13 +19,34 @@ api_response_t *api_response_create(
 
 
 void api_response_free(api_response_t *api_response) {
-	listEntry_t *listEntry;
-
-	free(api_response->code)
 
 	free(api_response->type)
 
 	free(api_response->message)
 
 	free(api_response);
 }
+
+cJSON *api_response_convertToJSON(api_response_t *api_response) {
+	cJSON *item = cJSON_CreateObject();
+	// api_response->code
+	if(cJSON_AddNumberToObject(item, "code", api_response->code) == NULL) {
+    	goto fail;
```

Comment:
why use goto? Why not insert the "cJSON_Delete(item);" and "return NULL;" in the if statement?

Reply:
also there seems to be an indentation error here.

Reply:
What do you want to refactor there? The checks are made at each method that allocated new memory to avoid a Segmentation Fault, they can't be combined. By combining and only doing the checks once you will already have a Segmentation Fault before the combined checks. There could only be written a new method that handles the cJSON_Delete(item); return NULL; and would be used somewhat like this:

if(cJSON_AddNumberToObject(item, "code", api_response->code) == NULL) {
    	return cJSON_deleteJSONObjectAndReturnNull(item);
...
cJSON *cJSON_deleteJSONObjectAndReturnNull(cJSON *item) {
    cJSON_Delete(item);
    return NULL;
}

This is unnecessary complicated and the goto is a lot easier and cleaner to read in my opinion. I really don't see a point to remove the goto's there as they are also used like that in the example code of the library.

Reply:
I think we can go with the current implementation for the beta release and consider @stkrwork suggestion as a day-2 requirement for improvement to the auto-generated C code.

Reply:
@wing328 
I agree. The community could have the chance to check and add PR to improve.